tests/integration/pilot/mirror_test.go

```diff
@@ -185,59 +175,86 @@ func runMirrorTest(options mirrorTestOptions) {
 				Inject: true,
 			})
 
-			var instances [3]echo.Instance
-			echoboot.NewBuilderOrFail(options.t, ctx).
-				With(&instances[0], echoConfig(ns, "a")). // client
-				With(&instances[1], echoConfig(ns, "b")). // target
-				With(&instances[2], echoConfig(ns, "c")). // receives mirrored requests
-				BuildOrFail(options.t)
-
-			if options.fnInjectConfig != nil {
-				options.fnInjectConfig(ns, ctx, instances)
+			builder := echoboot.NewBuilderOrFail(options.t, ctx)
```

Comment:
Would it help if we separated out the cluster from the initial `echo.Config`? We could have a number of clusters associated with the `Builder` and then add helpers to make working with it easier:

```golang
result := builder.echoboot.NewBuilderOrFail(options.t, ctx).
  // Apply these configs to all clusters in the environment
  // (maybe this is the default?)
  WithClusters(ctx.Environment().Clusters()).
  With(echoConfig(ns, "a").
  With(echoConfig(ns, "b").
  With(echoConfig(ns, "c").
  BuildOrFail(options.t)

// Return all of the "a" services indexed by the cluster.
allA := result.LookupByName("a")

// Return a single "a" Instance in the given cluster.
a0 := result.GetServiceOrFail(ctx, "a", ctx.Environment().Clusters()[0])

// Other methods?
```

WDYT?

Reply:
In a lot of scenarios that is great, but sometimes we want to test traffic to specific clusters. For that we have to make the service names unique to the cluster in question.

Reply:
@nmittler Check out the changes to echo builder; let me know if you have a better idea than SetupFn. I used that to handle the unique naming and setting the correct pilot instance. 

On that note, echo doesn't really use the pilot instance we pass after removing the native environment. Theres a lot that could be cleaned up if we just remove that from echo. WDYT?